### PR TITLE
[PROTOCOL RFC] Column Mapping Usage Tracking

### PR DESCRIPTION
**Pull request by tomvanbussel**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [x] Other (fill in here)

## Description

Adds the proposal for spec change Column Mapping Usage Tracking (see https://github.com/delta-io/delta/issues/2682) to the RFC folder.

## How was this patch tested?

N/A

## Does this PR introduce _any_ user-facing changes?

N/A

### CODE REVIEW

File: protocol_rfcs/column-mapping-usage-tracking.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+--------
+
+> New subsection at the end of the `Writer Requirements for Column Mapping` subsection
+
+### Writer Requirements for Usage Tracking
+
+In order to support column mapping usage tracking, writers must:
+- Write `protocol` and `metaData` actions when Column Mapping Usage Tracking is turned on for the first time:
+    - Write a `protocol` action with writer version 7 and the feature `columnMappingUsageTracking` in the `writerFeatures`.
+    - Write a `metaData` actions with the table property `delta.columnMapping.hasDroppedOrRenamed` set to `false` when creating a new table, or set to `true` when enabling usage tracking on an existing table.
```
[the remark this answers is not in the document]
Comment (tomvanbussel) [reply protocol_rfcs/column-mapping-usage-tracking.md]: Thanks!

File: protocol_rfcs/column-mapping-usage-tracking.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+--------
+
+> New subsection at the end of the `Writer Requirements for Column Mapping` subsection
+
+### Writer Requirements for Usage Tracking
+
+In order to support column mapping usage tracking, writers must:
+- Write `protocol` and `metaData` actions when Column Mapping Usage Tracking is turned on for the first time:
+    - Write a `protocol` action with writer version 7 and the feature `columnMappingUsageTracking` in the `writerFeatures`.
+    - Write a `metaData` action with the table property `delta.columnMapping.hasDroppedOrRenamed` set to `false` when creating a new table, or set to `true` when enabling usage tracking on an existing table.
```
Comment (felipepessoto) [inline protocol_rfcs/column-mapping-usage-tracking.md:41]: Enabling Usage Tracking for existing tables where column mapping is disabled also requires setting `delta.columnMapping.hasDroppedOrRenamed` to true?

Comment (felipepessoto) [reply protocol_rfcs/column-mapping-usage-tracking.md:41]: @tomvanbussel

File: protocol_rfcs/README.md [hunk 3]
```diff
@@ -16,12 +16,13 @@ Here is the history of all the RFCs propose/accepted/rejected since Feb 6, 2024,
...
-| 2023-02-28    | [vacuum-protocol-check.md](https://github.com/delta-io/delta/blob/master/protocol_rfcs/vacuum-protocol-check.md) | https://github.com/delta-io/delta/issues/2630 | Enforce Vacuum Protocol Check |
+| Date proposed | RFC file                                                                                                                          | Github issue                                  | RFC title                     |
+|:--------------|:----------------------------------------------------------------------------------------------------------------------------------|:----------------------------------------------|:------------------------------|
+| 2023-02-02    | [in-commit-timestamps.md](https://github.com/delta-io/delta/blob/master/protocol_rfcs/in-commit-timestamps.md)                    | https://github.com/delta-io/delta/issues/2532 | In-Commit Timestamps          |
+| 2023-02-09    | [type-widening.md](https://github.com/delta-io/delta/blob/master/protocol_rfcs/widening.md)                                       | https://github.com/delta-io/delta/issues/2623 | Type Widening                 |
```
Comment (felipepessoto) [inline protocol_rfcs/README.md:22]: If you don't mind, could you include this small fix? https://github.com/delta-io/delta/pull/2694

**Review by felipepessoto — APPROVED** [review]

**Review by felipepessoto — APPROVED** [review]

**Review by tdas — APPROVED** [review]

**Comment by tdas** [comment]
I am going to ignore the spark test failure (probably some flakiness) as this PR does not touch code.

**Pull request merged**